doc/modules.md

```diff
@@ -0,0 +1,88 @@
+Modularization
+============
+
+This document describes the changes for a more fine grained modularization
+of common libraries, e.g., Lunchbox
+
+## Requirements
+
+* Per-functionality targets
+* Improved build times for super projects
+
+## Implementation
+
+* All sub project targets are excluded from the default build, and
+  required targets are built only based on the dependencies.
+* Sub-libraries have a top-level header, e.g., lunchbox/log.h, and a
+  sub-directory with the implementation and CMakeLists, e.g.,
+  lunchbox/log/log.h and .cpp. The new common_add_library() cmake module
+  auto-generates the top-level header. New sub libraries set
+  NAME_OMIT_LIBRARY_HEADER.
+* The old common_library is deprecated, and stays during the transition
+  period.
+* Optional sub-targets are to be avoided and hidden by facades (esp. in
+  Lunchbox, see PersistentMap). If they are necessary, e.g, in hwsd,
+  downstream code needs to test for the target's availability (as done
+  e.g. in eq/server).
+* Servus will eventually be reintegrated as a lunchbox sub library.
+
+## Naming
+
+    common_add_library(Lunchbox::Log HEADERS logImpl.h PUBLIC_HEADERS log.h
+      SOURCES log.cpp DEPENDS PUBLIC Lunchbox::Bar PRIVATE Lunchbox::Foo)
+
+Generates: .../include/lunchbox/log.h, .../include/lunchbox/log/log.h,
+.../lib/libLunchboxLog.so
+
+    common_add_library(Lunchbox DEPENDS PUBLIC Lunchbox::Log)
+
+Generates: .../include/lunchbox.h
+
+## Examples
+
+    #include <zeroeq.h>
+    #include <lunchbox/log.h>
+
+### Desired Lunchbox Granularity
+
+    Lunchbox::Any
+    Lunchbox::Atomic
+    Lunchbox::Attic
+    Lunchbox::Bits
+    Lunchbox::Buffer
+    Lunchbox::Clock
+    Lunchbox::Condition
+    Lunchbox::DSO
+    Lunchbox::File
+    Lunchbox::Future
+    Lunchbox::Launcher
+    Lunchbox::Lock (lock, spinlock, scoped, monitor, queue, lockable)
+    Lunchbox::LockFree
+    Lunchbox::Log
+    Lunchbox::MPI
+    Lunchbox::MemoryMap
+    Lunchbox::Monitor
+    Lunchbox::OMP
+    Lunchbox::OS
+    Lunchbox::PersistentMap
+    Lunchbox::Plugin
+    Lunchbox::Process (daemon+launcher)
+    Lunchbox::RNG
+    Lunchbox::Servus
+    Lunchbox::TLS
+    Lunchbox::Test
+    Lunchbox::Thread
+    Lunchbox::URI
+    Lunchbox::UnorderedIntervalSet
+
+## Issues
+
+### 1: How do handle optional targets?
+
+_Resolution: Avoid them as possible, downstream projects need to check
+with if(TARGET) in cmake:_
+
+We could (re)use the COMPONENTS parameter in common_find_package(), but
```

Comment:
I would rather suggest to do like Qt and create one find script per target, which avoids the problem and allows much finer grained use in downstream projects. Real world example:

common_find_package(Qt5Core REQUIRED)
common_find_package(Qt5WebEngine 5.6)
if(NOT TARGET Qt5::WebEngine)
  common_find_package(Qt5WebKitWidgets SYSTEM)
endif()